test/integration/scheduler/framework_test.go

```diff
@@ -607,6 +632,55 @@ func TestReservePlugin(t *testing.T) {
 	}
 }
 
+// TestPostFilterPlugin tests invocation of post-filter plugins.
+func TestPostFilterPlugin(t *testing.T) {
+	// Create a plugin registry for testing. Register only a post-filter plugin.
+	registry := framework.Registry{postFilterPluginName: NewPostFilterPlugin}
+
+	// Setup initial post-filter plugin for testing.
+	pluginsConfig := &schedulerconfig.Plugins{
+		PostFilter: &schedulerconfig.PluginSet{
+			Enabled: []schedulerconfig.Plugin{
+				{
+					Name: postFilterPluginName,
+				},
+			},
+		},
+	}
+	// Set empty plugin config for testing
+	emptyPluginConfig := []schedulerconfig.PluginConfig{}
+
+	// Create the master and the scheduler with the test plugin set.
+	context := initTestSchedulerWithOptions(t,
+		initTestMaster(t, "post-filter-plugin", nil),
+		false, nil, registry, pluginsConfig, emptyPluginConfig, false, time.Second)
+	defer cleanupTest(t, context)
+
+	cs := context.clientSet
+	// Add a few nodes.
+	_, err := createNodes(cs, "test-node", nil, 2)
+	if err != nil {
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+
+	// Create a best effort pod.
+	pod, err := createPausePod(cs,
+		initPausePod(cs, &pausePodConfig{Name: "test-pod", Namespace: context.ns.Name}))
+	if err != nil {
+		t.Errorf("Error while creating a test pod: %v", err)
+	}
+
+	if err = waitForPodToSchedule(cs, pod); err != nil {
+		t.Errorf("Expected the pod to be scheduled. error: %v", err)
+	}
+
+	if postFilterPlugin.numPostFilterCalled == 0 {
+		t.Errorf("Expected the post-filter plugin to be called.")
+	}
+
```

Comment:
We should also test that when a post filter plugin returns error, the pod is not scheduled.

Reply:
Addressed